include/deal.II/base/tensor.h

```diff
@@ -2167,51 +2167,57 @@ Number determinant (const Tensor<1,1,Number> &t)
 
 
 /**
- * Cross-product in 2d. This is just a rotation by 90 degrees clockwise to
- * compute the outer normal from a tangential vector. This function is defined
- * for all space dimensions to allow for dimension independent programming
- * (e.g. within switches over the space dimension), but may only be called if
- * the actual dimension of the arguments is two (e.g. from the <tt>dim==2</tt>
- * case in the switch).
+ * Returns the cross-product in 2d. This is just a rotation by 90 degrees
+ * clockwise to compute the outer normal from a tangential vector. This
+ * function is defined for all space dimensions to allow for dimension
+ * independent programming (e.g. within switches over the space dimension),
+ * but may only be called if the actual dimension of the arguments is two
+ * (e.g. from the <tt>dim==2</tt> case in the switch).
  *
  * @relates Tensor
  * @author Guido Kanschat, 2001
  */
 template <int dim, typename Number>
 inline
-void
-cross_product (Tensor<1,dim,Number>       &dst,
-               const Tensor<1,dim,Number> &src)
+Tensor<1,dim,Number>
+cross_product (const Tensor<1,dim,Number> &src)
 {
   Assert (dim==2, ExcInternalError());
```

Comment:
This function only makes sense in dim==2. Can you try what happens if you only provide it for this particular template argument? (I imagine it was written this way to allow for generic code.)